perspective_reddit_bot/check_mod_actions.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A reddit bot to detect which actions subreddit moderators actually took."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import argparse
+from datetime import datetime
+import json
+import praw
+import time
+
+from creds import creds
+
+
+def write_moderator_actions(reddit,
+                            line,
+                            id_key,
+                            timestamp_key,
+                            output_path,
+                            hours_to_wait):
+  record = json.loads(line)
+  comment = reddit.comment(record[id_key])
+  maybe_wait(record[timestamp_key], hours_to_wait)
+  record['approved'] = comment.approved
+  record['removed'] = comment.removed
+  with open(output_path, 'a') as o:
+    json.dump(record, o)
+    o.write('\n')
+
+
+def maybe_wait(timestamp, hours_to_wait):
+  """Waits until hours_to_wait hours have passed since timestamp"""
+
+  now = datetime.utcnow()
+  time_diff = now - datetime.strptime(timestamp, '%Y%m%d_%H%M%S')
+  time_diff = time_diff.seconds
+  seconds_to_wait = hours_to_wait * 3600
+  if time_diff < seconds_to_wait:
+    time_to_wait = seconds_to_wait - time_diff
+    print('Waiting %.1f seconds...' % time_to_wait)
+    time.sleep(time_to_wait)
+
+
+def _main():
+  parser = argparse.ArgumentParser(
+      'Reads the output of moderate_subreddit.py and adds actions taken by'
+      'human moderators.')
+  parser.add_argument('input_path', help='json file with reddit comment ids')
+  parser.add_argument('output_path', help='path to write output file')
```

Comment:
Add checks that these arguments aren't empty? I think argparse has some features for declaring that flags should be set.

Reply:
I think because they don't start with '-' they are considered mandatory positional arguments:
https://docs.python.org/3/library/argparse.html#name-or-flags